lite/examples/sound_classification/android/app/src/main/java/org/tensorflow/lite/examples/soundclassifier/AudioBuffer.java

```diff
@@ -0,0 +1,152 @@
+package org.tensorflow.lite.examples.soundclassifier;
+
+import android.media.AudioFormat;
+import android.media.AudioRecord;
+import android.util.Log;
+
+import java.nio.FloatBuffer;
+
+public class AudioBuffer {
+
+    // TODO: What if the ring buffer is not yet fully filled before invocation?
+    private class FloatRingBuffer {
+        private float[] buffer;
+        private int current;
+
+        FloatRingBuffer(int size) {
+            buffer = new float[size];
+        }
+
+        public int getCapacity() {
+            return buffer.length;
+        }
+
+        public void feed(float v) {
+            buffer[current] = v;
+            current = (current + 1) % buffer.length;
+        }
+
+        public void feed(float[] data, int size) {
+            for (int i = 0; i < size; i++) {
+                feed(data[i]);
+            }
+        }
+
+        public float[] getArray() {
+            float[] output = new float[buffer.length];
+            for (int i = current; i < buffer.length; i++) {
+                output[i - current] = buffer[i];
+            }
+            for (int i = 0; i < current; i++) {
+                output[buffer.length - current + i] = buffer[i];
+            }
+            return output;
+        }
+    }
+
+    private AudioFormat audioFormat;
+    // Keeping all the data in float and convert them to desired types at the end.
+    // In the future, we might want to store data in the format specified by audioFormat.
+    private FloatRingBuffer floatRingBuffer;
+
+    // TODO: Do we need to keep track of the AudioFormat?
+    public AudioBuffer(AudioFormat audioFormat, int sampleCount) {
+        this.audioFormat = audioFormat;
+        this.floatRingBuffer = new FloatRingBuffer(sampleCount);
+    }
+
+    // PCM float
+    public int feed(float[] data) {
+        return feed(data, data.length);
+    }
+
+    // TODO: what's the correct name?
+    public int feed(float[] data, int size) {
+        floatRingBuffer.feed(data, size);
+        return size;
+    }
+
+    // PCM int16
+    public int feed(short[] data) {
+        return feed(data, data.length);
+    }
+
+    private float pcm16ToFloat(short v) {
+        return (float) v / 32768;
+    }
+
+    public int feed(short[] data, int size) {
+        for (int i = 0; i < size; i++) {
+            floatRingBuffer.feed(pcm16ToFloat(data[i]));
```

Comment:
Is using floatRingBuffer determined by the input AudioFormat when creating AudioBuffer? i.e. if the AudioFormat's encoding is ENCODING_PCM_16BIT, will we instantiate a ShortRingBuffer?